commandline/src/main/java/gov/cms/qpp/conversion/CommandLineRunner.java

```diff
@@ -0,0 +1,214 @@
+	 *
+	 * @param commandLine command line execute
+	 * @param fileSystem contextual file system to use for path operations
+	 */
+	public CommandLineRunner(CommandLine commandLine, FileSystem fileSystem) {
+		Objects.requireNonNull(commandLine, "commandLine");
+		Objects.requireNonNull(fileSystem, "fileSystem");
+
+		this.commandLine = commandLine;
+		this.fileSystem = fileSystem;
+	}
+
+	/**
+	 * Executes the command line
+	 */
+	@Override
+	public void run() {
+		if (isHelp()) {
+			sendHelp();
+		} else if (hasPotentialFiles()) {
+			Scopes scopes = getScopes();
+			scope = scopes.getQrdaScopes();
+			if (scopes.isValid()) {
+				Set<Path> convert = getRequestedFilesForConversion();
+
+				List<Path> invalid = convert.stream()
+						.filter(path -> !isValid(path))
+						.collect(Collectors.toList());
+				if (invalid.isEmpty()) {
+					doValidation = !commandLine.hasOption(CommandLineMain.SKIP_VALIDATION);
+					doDefaults = !commandLine.hasOption(CommandLineMain.SKIP_DEFAULTS);
+					historical = commandLine.hasOption(CommandLineMain.BYGONE);
+
+					convert.parallelStream()
+						.map(ConversionFileWriterWrapper::new)
+						.peek(conversion -> conversion.setContext(createContext()))
+						.forEach(ConversionFileWriterWrapper::transform);
+				} else {
+					DEV_LOG.error("Invalid or missing paths: " + invalid);
+					sendHelpHint();
+				}
+			} else {
+				DEV_LOG.error("A given template scope was invalid");
+				sendHelpHint();
+			}
+		} else {
+			DEV_LOG.error("You must specify files to convert");
+			sendHelpHint();
+		}
+	}
+
+	private Context createContext() {
+		Context context = new Context();
+		context.setDoDefaults(doDefaults);
+		context.setDoValidation(doValidation);
+		context.setHistorical(historical);
+		context.setScope(scope);
+		return context;
+	}
+
+	private boolean isHelp() {
+		return commandLine.hasOption(CommandLineMain.HELP);
+	}
+
+	private void sendHelp() {
+		StringWriter content = new StringWriter();
+		PrintWriter out = new PrintWriter(content);
+		CommandLineMain.HELP_FORMAT.printHelp(out, CommandLineMain.HELP_FORMAT.getWidth(),
+				"convert", null, CommandLineMain.OPTIONS, 0, 0, null, true);
+
+		DEV_LOG.info(content.toString());
+	}
+
+	private void sendHelpHint() {
+		DEV_LOG.info("Get help by rerunning with -help.");
+	}
+
+	private boolean hasPotentialFiles() {
+		return !commandLine.getArgList().isEmpty();
+	}
+
+	private Scopes getScopes() {
+		Scopes scopes = new Scopes();
+		if (commandLine.hasOption(CommandLineMain.TEMPLATE_SCOPE)) {
+			String[] templateScope = LITERAL_COMMA.split(commandLine.getOptionValue(CommandLineMain.TEMPLATE_SCOPE));
+			Set<QrdaScope> validScopes = Arrays.stream(templateScope)
+					.map(QrdaScope::getInstanceByName)
+					.filter(Objects::nonNull)
+					.collect(Collectors.toCollection(() -> EnumSet.noneOf(QrdaScope.class)));
+
+			if (validScopes.size() != templateScope.length) {
```

Comment:
How about `scopes.setValid(validScopes.size() == templateScope.length);` and scrap the if / else?

Reply:
Done